.github/workflows/deploy-vue-storefront-cloud.yml

```diff
@@ -47,63 +47,24 @@ jobs:
-            "front_version":"${{ github.sha }}",
-            "additional_apps": {
-              "apps":[{
-                "name":"middleware",
-                "tag":"${{ github.sha }}",
-                "image":"${{ secrets.DOCKER_REGISTRY_URL || 'registry.vuestorefront.cloud' }}/${{ secrets.PROJECT_NAME }}-storefrontcloud-io/vue-storefront-middleware",
-                "path":"/api/",
-                "port":"3000",
-                "has_base_path":false
-              }],
-              "enabled": true
-              }
-            }' ${{ secrets.CLOUD_API_BASE_URL || 'https://farmer.vuestorefront.cloud' }}/instance/${{ secrets.PROJECT_NAME }}-${{ secrets.CLOUD_REGION }}-gcp-storefrontcloud-io; then
-            echo "Instance updated"
-          else
-            echo "Something went wrong during the update process..."
-            exit 1
-          fi
+    - name: Deploy App
+      uses: fjogeleit/http-request-action@v1
+      with:
+        url: '${{ secrets.PLATFORM_API_URL }}/cloud/instances/${{ secrets.PROJECT_NAME }}-${{ secrets.CLOUD_REGION }}-gcp-storefrontcloud-io/deploy'
+        method: 'PATCH'
+        customHeaders: '{"Content-Type":"application/json"}'
+        data: '{ "cloudUserId": "${{ secrets.CLOUD_USERNAME }}", "cloudUserPassword": "${{ secrets.CLOUD_PASSWORD }}", "dockerImageHash": "${{ github.sha }}" }'
 
-      - name: Update deployment status (success)
-        if: success()
-        uses: chrnorm/deployment-status@releases/v1
-        with:
-          token: "${{ github.token }}"
-          target_url: https://${{ secrets.PROJECT_NAME }}.${{ secrets.CLOUD_REGION }}.gcp.storefrontcloud.io
-          state: "success"
-          description: Congratulations! The deploy is done.
-          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
-
-      - name: Update deployment status (failure)
-        if: failure()
-        uses: chrnorm/deployment-status@releases/v1
-        with:
-          token: "${{ github.token }}"
-          target_url: https://${{ secrets.PROJECT_NAME }}.${{ secrets.CLOUD_REGION }}.gcp.storefrontcloud.io
-          description: Unfortunately, the instance hasn't been updated.
-          state: "failure"
-          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+    - name: Deploy Middleware
+      uses: fjogeleit/http-request-action@v1
+      with:
+        url: '${{ secrets.PLATFORM_API_URL }}/cloud/instances/${{ secrets.PROJECT_NAME }}-${{ secrets.CLOUD_REGION }}-gcp-storefrontcloud-io/deploy'
+        method: 'PATCH'
+        customHeaders: '{"Content-Type":"application/json"}'
+        data: '{ "middleware":true, "cloudUserId":"${{ secrets.CLOUD_USERNAME }}", "cloudUserPassword":"${{ secrets.CLOUD_PASSWORD }}", "dockerImageHash":"${{ github.sha }}", "middlewareData":{ "path":"/api/", "port":3000, "has_base_path":false } }'
```

Comment:
Issue: Why do we have to all the time define path and port in every deployment?

Reply:
https://github.com/vuestorefront/platform-backend/pull/340#discussion_r1043098433

https://github.com/vuestorefront/platform-backend/pull/340#discussion_r1053751614

Reply:
But if you look at this, every time on every github action trigger, we put some data in this request, what is the purpose? - changing these parameters after every deployment? check is the same and fail? Could you explain it to me?

Reply:
This was done like this by Anastazy, as you can see:

https://github.com/vuestorefront/sample-express-application-with-middleware/commits/main

Here I don't really change anything, I just remove some things that in my view are static, and I leave them in the Platform/Console and I leave here the things that can be changed.

If in fact they are changed, how often, by whom, etc., I don't know. It was done that way, and I have no guidelines or context to change it, after all, this is a very critical part of the system. I don't believe that changing something without an explicit and important reason is the right thing to do.